### ISSUE TITLE
as user not working????

### ISSUE THREAD

**Issue by juanbrein** [capistrano/sshkit#361]
`#!/usr/bin/env ruby
require 'sshkit'                               
require 'sshkit/dsl'                           
include SSHKit::DSL                            
                                               
                                               
on %w{srv16} do |host|
  as 'jbrein' do
    execute 'whoami'                           
  end                                          
end                                            
`

Triggers:

`vagrant@srv16:`

where:

`
#!/usr/bin/env ruby
require 'sshkit'
require 'sshkit/dsl'
include SSHKit::DSL

on %w{srv16.infra.digitaltrends.com} do
  host.user = 'jbrein'
  execute 'whoami'
end
`

I get:

` INFO [d05e2de1] Running /usr/bin/env whoami as jbrein@srv16
  INFO [d05e2de1] Finished in 1.172 seconds with exit status 0 (successful).`

**Comment by mattbrictson** [comment]
Hi, thanks for the report. I think `as` works differently than you expect.

Using `as`, the connection is still established using the default user. In your case, the default user is `vagrant` (based on `$USER`, I assume).

It's only after the connection is established that `sudo -u [USER]` is used to run the command "as" that user.

If you want to change the user that is used for the actual SSH connection, then you should specify that user as part of the host using user@host syntax, like this:

```ruby
on %w{jbrein@srv16} do
  execute "whoami"
end
```

**Comment by juanbrein** [comment]
I see!  I was getting crazy trying to figure out what was going on. I don't see that in the README, can this be considered a bug in the documentation? Is it something that can lead to confusion or it was just me?

Thanks anyway for the clarification!

**Comment by mattbrictson** [comment]
Yes, I can imagine that would be confusing. Would you mind suggesting some edits? You can press the edit button (pencil icon) on either the [README](https://github.com/capistrano/sshkit/blob/master/README.md) or the [EXAMPLES](https://github.com/capistrano/sshkit/blob/master/EXAMPLES.md) and it will automatically create a pull request that I can review. Thanks!

**Comment by juanbrein** [comment]
Sure thing!

**Comment by leehambley** [comment]
bump @juanbrein - were you able to make some edits to the documentation to clarify this?

**Comment by juanbrein** [comment]
Sorry I'm on holidays, but will pick it up when back. Probably on Monday.
Thanks!
